mitylib/merge.py

```diff
@@ -1,288 +1,282 @@
-    # print(hc_samples)
-
-    mity_col_names = mity_col_names.split("\t")
-    mity_samples = mity_col_names[9:]
-    # print(mity_samples)
-
-    if hc_samples == mity_samples:
-        sys.stderr.write("samples in same order - can just add mity variants to hc vcf")
-        # add the hc variants with the mity variants - will be ordered later
-        hc_col_names = ("\t").join(hc_col_names)
-
-        new_mity_variants = []
-        for line in mity_variants:
-            new_line = ("\t").join(line)
-            new_mity_variants.append(new_line)
-
-        new_vcf = merged_header + [hc_col_names] + new_mity_variants + hc_variants
-        for line in new_vcf:
-            print(line)
-    else:
-        # check that they have the same samples
-        if set(hc_samples) != set(mity_samples):
-            sys.exit("VCFs have different sample names")
-        mity_variants_no_sample = [x[:9] for x in mity_variants]
-        mity_variants_sample = [x[9:] for x in mity_variants]
-        # print(mity_variants_sample)
-        # sys.stderr.write("samples not in the same order - will reorder mity
-        # variants")
-
-        # get the order of the hc sampels
-        # print("hc samples")
-        # print(hc_samples)
-        # print("mity samples")
-        # print(mity_samples)
-        # mity_samples_idx = [hc_samples.index(x) for x in mity_samples]
-        mity_samples_idx = [mity_samples.index(x) for x in hc_samples]
-
-        # print(mity_samples_idx)
-        # sys.exit()
-        mity_variants_reordered_sample = []
-        for line in mity_variants_sample:
-            mity_variants_reordered_sample.append([line[x] for x in mity_samples_idx])
-        # print(mity_variants_reordered_sample)
-        # sys.exit()
-        # add the newly ordered samples back onto the mity_variants_no_sample
-        new_mity_variants = []
-        for i in range(0, len(mity_variants_no_sample)):
-            new_line = mity_variants_no_sample[i] + mity_variants_reordered_sample[i]
-            # print(new_line)
-            new_line = ("\t").join(new_line)
-            new_mity_variants.append(new_line)
-
-        # add the hc variants with the mity variants - will be ordered later
-        hc_col_names = ("\t").join(hc_col_names)
-        new_vcf = merged_header + [hc_col_names] + new_mity_variants + hc_variants
-
-        # write_merged_vcf(new_vcf, outfile, genome)
+            section = "INFO"
+
+        line = line.split(",")[0]
+        field_id = line.split("=")[-1]
+
+        return section, field_id
+
+    def get_header_description(self, line: str):
+        """
+        Get header description.
+        """
+        return line.split("Description=")[1].strip('>" ')
+
+    def make_new_line(self, nuclear_line, mity_file_line_num):
+        """
+        Makes an updated line with merged description.
+        """
+        with gzip.open(self.mity_vcf_path, "rt") as file:
+            file.seek(mity_file_line_num)
```

Comment:
What's the behaviour if you see to a line that does not exist and then try to read it?

Reply:
From testing, `file.readline()` just returns an empty string but afaik it's undefined behaviour. It only takes seek inputs from `tell()` which reads the position from the same file, so it shouldn't ever be out of bounds.